src/pages/DragonPage/DragonsSyrup.tsx

```diff
@@ -130,27 +131,18 @@ const DragonsSyrup: React.FC = () => {
     [sortedSyrupInfos],
   );
 
-  useEffect(() => {
-    setSyrupInfos(undefined);
-    setTimeout(() => setPageLoading(false), 500); //load syrups 0.5s after loading page
-  }, []);
-
   useEffect(() => {
     setPageIndex(0);
-    setSyrupInfos(sortedSyrupInfos.slice(0, LOADSYRUP_COUNT));
-    return () => setSyrupInfos(undefined);
-    // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [syrupRewardAddress]);
 
-  useEffect(() => {
-    const currentSyrupInfos = syrupInfos || [];
-    const syrupInfosToAdd = sortedSyrupInfos.slice(
-      currentSyrupInfos.length,
-      currentSyrupInfos.length + LOADSYRUP_COUNT,
-    );
-    setSyrupInfos(currentSyrupInfos.concat(syrupInfosToAdd));
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [pageIndex]);
+  const syrupInfos = useMemo(() => {
+    return sortedSyrupInfos
+      ? sortedSyrupInfos.slice(
+          0,
+          getPageItemsToLoad(pageIndex, LOADSYRUP_COUNT),
+        )
```

Comment:
This should just be an extension on an array. E
G

sortedSyrupInfos.getPage(index, count)

Reply:
I did not understood this please explain in more detail,

are you talking about creating array prototype method

Reply:
Yeah you can create an extend the prototype